src/main/webapp/src/app/hidden-gem.service.spec.ts

```diff
@@ -0,0 +1,55 @@
+import { HiddenGemService } from "./hidden-gem.service";
+import { HiddenGem } from './hidden-gem';
+import { HttpErrorResponse } from '@angular/common/http';
+import { of } from 'rxjs';
+import { fakeAsync } from '@angular/core/testing';
+
+let httpClientSpy: { get: jasmine.Spy };
+let hiddenGemService: HiddenGemService;
+
+beforeEach(() => {
+  httpClientSpy = jasmine.createSpyObj('HttpClient', ['get']);
+  hiddenGemService = new HiddenGemService(httpClientSpy as any);
+});
+
+it('should return expected hidden gems (HttpClient called once)', () => {
+  const expectedHiddenGems: HiddenGem[] =
+    [ { id: 1, name: 'Fratelli Fresh', business_type: 'restaurant', address: 'ICC Sydney, tenancy 2/14 Darling Dr, Sydney NSW 2000', price_level: 2, rating: 3.6, photo: 'https://cdn.pixabay.com/photo/2017/12/09/08/18/pizza-3007395_960_720.jpg' },
```

Comment:
nit: I think this would be more readable if you split this over several lines eg:
```
[{ id: 1,
   name: 'Fratelli Fresh',
   // etc...
```